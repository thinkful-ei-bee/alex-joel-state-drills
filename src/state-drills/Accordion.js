import React, {Component} from 'react';

class Accordion extends Component{

    state={
        currSection:null,
    }

    handleButtonClick= (index) => {
        //console.log(index);
        this.setState({
            currSection:index,
        })
    }

    render(){
        const thisSection = this.state.currSection;
        //console.log(this.props.sections);
        const listItems = this.props.sections.map((section,index)=>
                (
                    <li>
                        <button onClick={() => this.handleButtonClick(index)}>
                            {section.title}
                        </button>
                        {thisSection===index && <p>{section.content}</p>}
                    </li>
                ))
        return(
            <ul className=''>
               {listItems}
            </ul>
        )
    }
};

export default Accordion;