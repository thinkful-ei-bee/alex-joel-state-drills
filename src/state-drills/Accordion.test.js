import React from 'react';
import Accordion from './Accordion';
import ReactDOM from 'react-dom';
import {sections} from './AccordionStore';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe('Accordion Test', () =>{
    it('Accordion renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion sections={sections}/>, div);
        ReactDOM.unmountComponentAtNode(div);
      });

    it('Accordion renders empty without prop',() =>{
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders no active sections', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })

      it('renders a clicked section', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
      })

      it('renders only last section when multiple have been clicked', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
      })
})
