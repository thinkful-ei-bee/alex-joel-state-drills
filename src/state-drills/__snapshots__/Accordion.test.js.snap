// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Test Accordion renders empty without prop 1`] = `
<ul
  className="listOfSections"
/>
`;

exports[`Accordion Test renders a clicked section 1`] = `
<ul
  className="listOfSections"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Test renders no active sections 1`] = `
<ul
  className="listOfSections"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Test renders only last section when multiple have been clicked 1`] = `
<ul
  className="listOfSections"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;
